boring-sys/build.rs

```diff
@@ -288,6 +288,14 @@ fn get_extra_clang_args_for_bindgen() -> Vec<String> {
             params.push("-isysroot".to_string());
             params.push(sysroot);
         }
+        "android" => {
+            let android_ndk_home = std::env::var("ANDROID_NDK_HOME")
+                .expect("Please set ANDROID_NDK_HOME for Android build");
```

Comment:
This is now required when setting `BORING_BSSL_PATH` to use a pre-compiled boringssl object file. Is that intentional? It seems reasonable for these requirements to match, but I don't use Android so I don't know how necessary the NDK is for compiling code.

Reply:
That's an interesting point. Another way to phrase this is "how much do the system C headers affect BoringSSL's public headers" and I haven't looked into that at all. C is a finicky language, but I'd also expect BoringSSL's public API surface to be pretty, well, boring.

Reply:
I will say that the further the host platform is from the Android platform being targeted, though, the more likely there'll be a problem.

Reply:
ah I see, the way this can go wrong is something like
```c
 int ASN1_TYPE_get(const ASN1_TYPE *a);
```
and `int` has a different size depending on whether you're using the Android SDK or not. Ok, I agree we should require this even if it leads to build failures in rare cases; better that than to silently use the wrong ABI.

Reply:
Right. Usually even that will be fine because bindgen will generate `libc::c_int`, but that's the type of thing that could get messed up.

Reply:
makes sense - I worry most about it in macro expressions, bindgen's handling of `#define x y` is ... pretty hacky.